client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/context/FlexibleExecutorContext.java

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright 2014-2016 CyberVision, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kaaproject.kaa.client.context;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.SynchronousQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+public class FlexibleExecutorContext extends AbstractExecutorContext implements ExecutorContext {
+  private static final Logger LOG = LoggerFactory.getLogger(FlexibleExecutorContext.class);
+
+  private static final int DEFAULT_MAX_THREAD_IDLE_MILLISECONDS = 100;
+  private static final int DEFAULT_MAX_THREADS = Integer.MAX_VALUE;
+
+  private final int lifeCycleThreadCount;
+  private final int apiThreadCount;
+  private final int callbackThreadCount;
+  private final int scheduledThreadCount;
```

Comment:
Here you should better use naming in style "maxLifeCycleThreads" instead of style "lifeCycleThreadsCount". Such the naming will better illustrate variables meaning.